include/deal.II/fe/fe_tools.h

```diff
@@ -977,6 +977,38 @@ namespace FETools
                           const bool do_tensor_product = true);
 
     /**
+     * Same as above for an arbitrary number of parameters of type
+     * <code>std::pair<std::unique_ptr<FiniteElement<dim, spacedim>>, unsigned int></code>
+     * and <code>do_tensor_product = true</code>.
+     */
+    template <int dim, int spacedim, class... FESystems,
+              typename = typename std::enable_if<all_same_as<typename std::decay<FESystems>::type...,
+                                                             std::pair<std::unique_ptr<FiniteElement<dim, spacedim>>,
+                                                                 unsigned int>>::value>::type>
+    FiniteElementData<dim>
+    multiply_dof_numbers (FESystems &&... fe_systems)
+    {
+      std::vector<const FiniteElement<dim,spacedim>*> fes;
+      std::vector<unsigned int> multiplicities;
+
+      const auto extract =
+        [&fes, &multiplicities]
+        (const std::pair<std::unique_ptr<FiniteElement<dim, spacedim>>, unsigned int> &fe_system)
+      {
+        fes.push_back(fe_system.first.get());
+        multiplicities.push_back(fe_system.second);
+      };
+      [](...) {}((extract(std::forward<FESystems>(fe_systems)),0)...);
```

Comment:
same issue here, as well as the question of why the order is wrong

same for the functions below

Reply:
Similarly adressed.